src/PIL/Image.py

```diff
@@ -1155,10 +1155,7 @@ def convert_transparency(m, v):
         if trns is not None:
             if new_im.mode == "P":
                 try:
-                    new_im.info["transparency"] = new_im.palette.getcolor(
-                        cast(Tuple[int, int, int], trns),  # trns was converted to RGB
-                        new_im,
-                    )
+                    new_im.info["transparency"] = new_im.palette.getcolor(trns, new_im)
```

Comment:
I'm not too sure about this change.
Although mypy no longer complains, I can see a warning in PyCharm.
Running mypy with `--strict` prints a warning that could explain why it (incorrectly) does not error without the cast:
```
src\PIL\Image.py:1158: error: Item "None" of "Optional[Any]" has no attribute
"getcolor"  [union-attr]
    ...                   new_im.info["transparency"] = new_im.palette.getcol...
                                                        ^~~~~~~~~~~~~~~~~~~~~~~
```
I suspect that the warning is only skipped because mypy doesn't realize that `new_im.palette` is an `ImagePalette` because it is set within an if block (unlike the cast above which is not in a nested if block).

Reply:
I've created https://github.com/python-pillow/Pillow/pull/8167, where I give up on the idea of requiring `getcolor()` to have a length of exactly 3 or 4. If that is merged, then my next suggestion would be https://github.com/radarhere/Pillow/commit/dfc1b0f9aec38d0b4e02ae9095efb1229f32bbd4

But I am keen to get https://github.com/python-pillow/Pillow/pull/8032 in for this release, so at any stage if you say 'Hey, let's merge it as it is and debate this later', we absolutely can.

Reply:
Both the PRs have been merged, so I've created https://github.com/python-pillow/Pillow/pull/8169